src/cljs/ataru/virkailija/routes.cljs

```diff
@@ -44,9 +44,16 @@
   (defroute #"/editor/(\d+)" [id]
     (dispatch [:set-active-panel :editor])
     (dispatch [:editor/refresh-forms])
-    (dispatch [:editor/select-form (js/parseInt id 10)]))
+    (dispatch-after-state
+      :predicate
+      (fn [db] (not-empty (get-in db [:editor :forms])))
+      :handler
+      (fn [forms]
+        (dispatch [:editor/select-form (js/parseInt id 10)]))))
```

Comment:
Maybe use the more robust Number-function instead? parseInt allows garbage:

cljs.user=> (js/parseInt "666AkiKaivola" 10)
666
cljs.user=> (js/Number "666AkiKaivola")
NaN
cljs.user=> (js/Number "44")
44